src/evaluate/evaluator.py

```diff
@@ -12,95 +12,125 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from abc import ABC, abstractmethod
 from numbers import Number
-from typing import Dict, Optional, Union
+from typing import Any, Callable, Dict, List, Optional, Tuple, Union
 
 # Lint as: python3
-import numpy as np
-from datasets import Dataset, DatasetDict, IterableDataset, IterableDatasetDict, load_dataset, load_metric
+from datasets import Dataset, load_dataset, load_metric
 from datasets.metric import Metric
```

Comment:
`evaluate` is replacing `datasets.load_metric` and `Metric`. For `load_metric` you can use `evaluate.load` and the new class for `Metric` is `EvaluationModule`. Sorry, didn't catch this earlier!